basesandbox/src/ipc/mod.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub mod domain_socket;
+pub mod intra;
+pub mod semaphore;
+use std::collections::hash_map::HashMap;
+
+/// Inter-process obejct that is constructed / synchronized by another Inter-process mechanism.
+pub trait InterProcessUnit {
```

Comment:
Separation of `InterProcessUnit` trait seems overkill. Why a separate trait while `Ipc` seems enough? Too fine-grained abstraction might make codes difficult to grok.

Reply:
The only example that is `InterProcessUnit` but not IPC was the semaphore.
Since I'll remove that, I'll remove this too.